### PR TITLE
#509 allow tokens to be used in credentials

### PR DESCRIPTION
**Pull request by JonCanning**
This introduces the concept of `<token>` as an alternative to `<credential>` in the config

`paket config add-token githubAuthKey token`

### CODE REVIEW

File: src/Paket.Core/ConfigFile.fs [hunk 1]
```diff
@@ -88,14 +88,17 @@ let private readPassword (message : string) : string =
...
-    let username = node.Attributes.["username"].Value
-    let password = node.Attributes.["password"].Value
-    let salt = node.Attributes.["salt"].Value
-
-    username, Decrypt salt password
-
-let private createSourceNode (credentialsNode : XmlNode) source =
-    let node = credentialsNode.OwnerDocument.CreateElement("credential")
+    match node.Name.ToLowerInvariant() with
+    | "credential" ->
```
Comment (forki) [inline src/Paket.Core/ConfigFile.fs]: is this backwards-compatible?

**Comment by forki** [comment]
checkout

==> very cool stuff!


could you please add a bit of docs abot the two modes? maybe at http://fsprojects.github.io/Paket/github-dependencies.html#Referencing-a-private-github-repository?

thx.

File: src/Paket.Core/ConfigFile.fs (hunk 1 above)
Comment (JonCanning) [inline src/Paket.Core/ConfigFile.fs]: It is for me but you've got me worried now. Not sure how else to test it.

Comment (forki) [inline src/Paket.Core/ConfigFile.fs]: No I guess it is. But I don't know that part of the code very well so I was
just asking.

**Pull request merged**